core/src/main/scala/cats/data/package.scala

```diff
@@ -1,33 +1,14 @@
 package cats
 
 package object data {
-  type NonEmptyList[A] = OneAnd[List, A]
   type NonEmptyStream[A] = OneAnd[Stream, A]
   type ValidatedNel[E, A] = Validated[NonEmptyList[E], A]
 
-  def NonEmptyList[A](head: A, tail: List[A] = Nil): NonEmptyList[A] =
-    OneAnd(head, tail)
-  def NonEmptyList[A](head: A, tail: A*): NonEmptyList[A] =
-    OneAnd[List, A](head, tail.toList)
-
   def NonEmptyStream[A](head: A, tail: Stream[A] = Stream.empty): NonEmptyStream[A] =
     OneAnd(head, tail)
   def NonEmptyStream[A](head: A, tail: A*): NonEmptyStream[A] =
     OneAnd(head, tail.toStream)
 
-  object NonEmptyList {
-    def fromReducible[F[_], A](fa: F[A])(implicit F: Reducible[F]): Eval[NonEmptyList[A]] =
```

Comment:
@kailuowang I agree that there should be a `toNonEmptyVector`. I think I'd like to opt to put that into a follow-up PR though.

As far as a type class for these conversions, do you have any ideas for improvements above `Reducible`? I like `Reducible` because it is a pretty general abstraction.